pkg/liqo-controller-manager/networking/internal-network/route/internalnode_controller.go

```diff
@@ -155,20 +172,29 @@ func (r *InternalNodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&networkingv1beta1.InternalNode{}).
 		Watches(&networkingv1beta1.Configuration{}, handler.EnqueueRequestsFromMapFunc(r.genericEnqueuerfunc)).
 		Watches(&ipamv1alpha1.IP{}, handler.EnqueueRequestsFromMapFunc(r.genericEnqueuerfunc)).
+		Watches(&corev1.Node{}, handler.EnqueueRequestsFromMapFunc(r.genericEnqueuerfunc)).
 		Complete(r)
 }
 
-func (r *InternalNodeReconciler) genericEnqueuerfunc(ctx context.Context, _ client.Object) []reconcile.Request {
+func (r *InternalNodeReconciler) genericEnqueuerfunc(ctx context.Context, obj client.Object) []reconcile.Request {
 	internalNodes, err := getters.ListInternalNodesByLabels(ctx, r.Client, labels.Everything())
 	if err != nil {
 		klog.Error(err)
 		return nil
 	}
+
+	_, isNode := obj.(*corev1.Node)
+
 	var requests []reconcile.Request
 	for i := range internalNodes.Items {
-		requests = append(requests, reconcile.Request{
-			NamespacedName: client.ObjectKeyFromObject(&internalNodes.Items[i]),
-		})
+		iNode := &internalNodes.Items[i]
+
+		// When the object that triggered the watch is a Node, enqueue the event only for the InternalNode related to that Node.
+		if isNode && iNode.Name == obj.GetName() {
+			requests = append(requests, reconcile.Request{
+				NamespacedName: client.ObjectKeyFromObject(iNode),
+			})
+		}
 	}
 	return requests
 }
```

Comment:
I would do a custom enqueuer for the node Watch(), to avoid casting and reduce logic complexity